### DISCUSSION TITLE
How to set up Docker?

### DISCUSSION THREAD

**Discussion by OfficialCRUGG** [mastodon/mastodon#23556] [Q&A]
Hello there,

I couldn't find an up-to-date guide on how to install Mastodon in Docker.

I found this guide, but it is relatively outdated and doesn't work for me:
https://github.com/felx/mastodon-documentation/blob/master/Running-Mastodon/Docker-Guide.md

Besides now having to use `docker compose` instead of `docker-compose`, it always fails when `docker compose run --rm web rake mastodon:setup` asks me to enter the PostgreSQL data. When I try to enter the `db` host name, it says that it cannot translate that host name to an address, even though `db` is clearly defined as a service in the default docker-compose file.

Is there any up to date docker installation guide? Or can anyone provide me with more details on how to get the PostgreSQL setup working?

**Comment by xiconfjs** [comment]
this is the best post I found so far:

https://goneuland.de/mastodon-mit-docker-und-traefik-installieren/
